### DISCUSSION TITLE
Unable to run tests locally?

### DISCUSSION THREAD

**Discussion by larsks** [netbox-community/netbox#11426] [General]
In preparation for submitting a pull request, I'd like to run the tests locally, but I'm running into some difficulties.  I'm trying to run them like this:

```
export NETBOX_CONFIGURATION=netbox.configuration_Test
python netbox/manage.py test netbox/
```

This produces:

```
Found 5314 test(s).
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
E
```

And then it gets stuck there forever. I end up with a Python process pegged at 100% cpu:

```
    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
2247976 lars      20   0 1790588   1.5g  19432 R  93.8   2.4   9:18.07 python netbox/manage.py test netbox/
```

I see the same behavior even if I limit the tests like this:

```
python netbox/manage.py test netbox/ -k test_remote_auth
```

This only discovers 10 tests but still gets stuck.

Has anyone seen this behavior?

**Comment by kkthxbye-code** [comment]
Can you try to pass -v 3 to the test command to get some more verbose output?

**Reply by larsks** [reply]
Running `python netbox/manage.py test -v3 netbox/ -k test_remote_auth` it gets stuck after:

```
.
.
.
Adding content type 'dummy_plugin | dummymodel'
Adding permission 'Permission object (458)'
Adding permission 'Permission object (459)'
Adding permission 'Permission object (460)'
Adding permission 'Permission object (461)'
System check identified no issues (0 silenced).
extras.tests.test_api (unittest.loader._FailedTest.extras.tests.test_api) ... ERROR
```

If I `strace` the process all I see are `mmap` and `munmap` calls, which doesn't really tell me anything useful.
